File: CSharp/Plugins/UniversalEditor.Essential/DataFormats/Markup/XML/XMLDataFormatSettings.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace UniversalEditor.DataFormats.Markup.XML
{
	public class XMLDataFormatSettings
	{
		private char mvarTagBeginChar = '<';
		private char mvarTagSpecialDeclarationStartChar = '!';
		private string mvarTagSpecialDeclarationCommentStart = "--";
		private char mvarTagEndChar = '>';
		private char mvarTagCloseChar = '/';
		private char mvarPreprocessorChar = '?';
		private char mvarTagNamespaceSeparatorChar = ':';
		private char mvarAttributeNameValueSeparatorChar = '=';
		private char mvarEntityBeginChar = '&';
		private char mvarEntityEndChar = ';';
		public char TagBeginChar
		{
			get
			{
				return this.mvarTagBeginChar;
			}
			set
			{
				this.mvarTagBeginChar = value;
			}
		}
		public char TagSpecialDeclarationStartChar
		{
			get
			{
				return this.mvarTagSpecialDeclarationStartChar;
			}
			set
			{
				this.mvarTagSpecialDeclarationStartChar = value;
			}
		}
		public string TagSpecialDeclarationCommentStart
		{
			get
			{
				return this.mvarTagSpecialDeclarationCommentStart;
			}
			set
			{
				this.mvarTagSpecialDeclarationCommentStart = value;
			}
		}
		public char TagEndChar
		{
			get
			{
				return this.mvarTagEndChar;
			}
			set
			{
				this.mvarTagEndChar = value;
			}
		}
		public char TagCloseChar
		{
			get
			{
				return this.mvarTagCloseChar;
			}
			set
			{
				this.mvarTagCloseChar = value;
			}
		}
		public char PreprocessorChar
		{
			get
			{
				return this.mvarPreprocessorChar;
			}
			set
			{
				this.mvarPreprocessorChar = value;
			}
		}
		public char TagNamespaceSeparatorChar
		{
			get
			{
				return this.mvarTagNamespaceSeparatorChar;
			}
			set
			{
				this.mvarTagNamespaceSeparatorChar = value;
			}
		}
		public char AttributeNameValueSeparatorChar
		{
			get
			{
				return this.mvarAttributeNameValueSeparatorChar;
			}
			set
			{
				this.mvarAttributeNameValueSeparatorChar = value;
			}
		}
		public char EntityBeginChar
		{
			get
			{
				return this.mvarEntityBeginChar;
			}
			set
			{
				this.mvarEntityBeginChar = value;
			}
		}
		public char EntityEndChar
		{
			get
			{
				return this.mvarEntityEndChar;
			}
			set
			{
				this.mvarEntityEndChar = value;
			}
		}

		private bool mvarIsStandalone = false;
		public bool IsStandalone { get { return mvarIsStandalone; } set { mvarIsStandalone = value; } }

		private System.Collections.Specialized.StringCollection mvarAutoCloseTagNames = new System.Collections.Specialized.StringCollection();
		public System.Collections.Specialized.StringCollection AutoCloseTagNames { get { return mvarAutoCloseTagNames; } }

		private Collections.Generic.BidirectionalDictionary<string, string> mvarEntities = new Collections.Generic.BidirectionalDictionary<string, string>();
		public Collections.Generic.BidirectionalDictionary<string, string> Entities { get { return mvarEntities; } }

		private char mvarCDataBeginChar = '[';
		public char CDataBeginChar { get { return mvarCDataBeginChar; } set { mvarCDataBeginChar = value; } }
		private char mvarCDataEndChar = ']';
		public char CDataEndChar { get { return mvarCDataEndChar; } set { mvarCDataEndChar = value; } }

		private bool mvarPrettyPrint = true;
		/// <summary>
		/// Determines whether to insert tabs and spaces to "pretty-print" the output XML.
		/// </summary>
		public bool PrettyPrint { get { return mvarPrettyPrint; } set { mvarPrettyPrint = value; } }
	}
}

File: Java/net.alcetech.UniversalEditor.Plugins.Core/src/net/alcetech/UniversalEditor/DataFormats/Markup/XML/XMLDataFormatSettings.java
package net.alcetech.UniversalEditor.DataFormats.Markup.XML;

import java.util.ArrayList;

import net.alcetech.ApplicationFramework.Collections.Generic.BidirectionalDictionary;
import net.alcetech.ApplicationFramework.Collections.Generic.KeyValuePair;

public class XMLDataFormatSettings {
	
	private BidirectionalDictionary<String, String> _entities = new BidirectionalDictionary<String, String>();
	public void addEntity(String name, String value) {
		_entities.add(name, value);
	}
	public String getEntityValue(String name) {
		return _entities.getValue2(name);
	}
	public String getEntityName(String value) {
		return _entities.getValue1(value);
	}
	
	private ArrayList<String> _autoCloseTagNames = new ArrayList<String>();
	public void addAutoCloseTagName(String name) {
		_autoCloseTagNames.add(name);
	}
	public boolean containsAutoCloseTagName(String name) {
		return _autoCloseTagNames.contains(name);
	}

	private char _TagBeginChar = '<';
	public char getTagBeginChar() { return _TagBeginChar; }
	public void setTagBeginChar(char value) { _TagBeginChar = value; }
	
	private char _TagCloseChar = '/';
	public char getTagCloseChar() { return _TagCloseChar; }
	public void setTagCloseChar(char value) { _TagCloseChar = value; }
	
	private char _TagEndChar = '>';
	public char getTagEndChar() { return _TagEndChar; }
	public void setTagEndChar(char value) { _TagEndChar = value; }
	
	private char _TagSpecialDeclarationStartChar = '!';
	public char getTagSpecialDeclarationStartChar() {
		return _TagSpecialDeclarationStartChar;
	}
	public void setTagSpecialDeclarationStartChar(char value) {
		_TagSpecialDeclarationStartChar = value;
	}
	
	private String _TagSpecialDeclarationCommentStart = "--";
	public String getTagSpecialDeclarationCommentStart() {
		return _TagSpecialDeclarationCommentStart;
	}
	public void setTagSpecialDeclarationCommentStart(String value) {
		_TagSpecialDeclarationCommentStart = value;
	}
	
	private char _PreprocessorChar = '?';
	public char getPreprocessorChar() {
		return _PreprocessorChar;
	}
	public void setPreprocessorChar(char value) {
		_PreprocessorChar = value;
	}
	
	private char _CDataBeginChar = '[';
	public char getCDataBeginChar() { return _CDataBeginChar; }
	public void setCDataBeginChar(char value) { _CDataBeginChar = value; }
	
	private char _CDataEndChar = ']';
	public char getCDataEndChar() { return _CDataEndChar; }
	public void setCDataEndChar(char value) { _CDataEndChar = value; }
	
	public KeyValuePair<String, String>[] getEntities() {
		return _entities.toArray();
	}
	
	private char _EntityBeginChar = '&';
	public char getEntityBeginChar() { return _EntityBeginChar; }
	public void setEntityBeginChar(char value) { _EntityBeginChar = value; }

	private char _EntityEndChar = ';';
	public char getEntityEndChar() { return _EntityEndChar; }
	public void setEntityEndChar(char value) { _EntityEndChar = value; }
	
	private char _AttributeNameValueSeparatorChar = '=';
	public char getAttributeNameValueSeparatorChar() { return _AttributeNameValueSeparatorChar; }
	public void setAttributeNameValueSeparatorChar(char value) { _AttributeNameValueSeparatorChar = value; }
}
